### PR TITLE
[WIP] Category fixes

### PR DESCRIPTION
**Pull request by craigh**
DO NOT MERGE YET

| Q                 | A
| ----------------- | ---
| Bug fix?          | yes
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Tests pass?       | yes
| Fixed tickets     | -
| Refs tickets      | #2206
| License           | MIT
| Doc PR            | n/a
| Changelog updated | no

refs #2206

ping @jusuff

### CODE REVIEW

File: src/lib/util/CategoryUtil.php [hunk 1]
```diff
@@ -968,6 +968,33 @@ public static function getCategoryTreeJqueryJS($cats, $doReplaceRootCat = true,
         return $tree->getJqueryHtml();
     }
 
+    public static function getJsTreeNodeFromCategory(\Zikula\Module\CategoriesModule\Entity\CategoryEntity $category)
```
Comment (cmfcmf) [inline src/lib/util/CategoryUtil.php]: Does this method need to be public?

Comment (craigh) [inline src/lib/util/CategoryUtil.php]: I believe it will be useful publicly, yes.

**Comment by craigh** [comment]
@Guite could use your help in testing this. I think it is done.

**Comment by Guite** [comment]
I tested the current branch trying each context action as well as drag n drop. Overall it looks you did a very great job :-)

Here are the only things I've found:

* Editing a category works, but the updated (localised) name is not reflected in the tree properly until the page is reloaded.
* Moving a category between two other ones or into another one results in a 500 error in the `resequence` method: `Error: Call to a member function setSort_value() on a non-object`

**Pull request merged**